astropy/stats/spatial.py

```diff
@@ -102,31 +117,36 @@ def x_max(self, value):
             )
 
     @property
-    def y_min(self):
+    def y_min(self) -> float | None:
         return self._y_min
 
     @y_min.setter
-    def y_min(self, value):
+    def y_min(self, value: float | None) -> None:
         if value is None or isinstance(value, (float, int)):
             self._y_min = value
         else:
             raise ValueError(f"y_min is expected to be a real number. Got {value}.")
 
     @property
-    def x_min(self):
+    def x_min(self) -> float | None:
         return self._x_min
 
     @x_min.setter
-    def x_min(self, value):
+    def x_min(self, value: float | None) -> None:
         if value is None or isinstance(value, (float, int)):
             self._x_min = value
         else:
             raise ValueError(f"x_min is expected to be a real number. Got {value}.")
 
-    def __call__(self, data, radii, mode="none"):
+    def __call__(
+        self,
+        data: NDArray,
```

Comment:
```suggestion
        data: NDArray[float]
```
?

Reply:
Yes I think that's right, `data` and `radii` should be `NDArray[float]`, as should the return type. Fixed throughout `spatial` in [d720e5c](https://github.com/astropy/astropy/pull/16562/commits/d720e5c540310de94e579b9f4a73cee5724b9dbd)